lib/explorer/data_frame.ex

```diff
@@ -5785,6 +5785,12 @@ defmodule Explorer.DataFrame do
   defimpl Inspect do
     import Inspect.Algebra
 
+    def inspect(df, _opts) when node(df.data.resource) != node() do
+      raise RuntimeError.exception(
+              "It is not possible to inspect a DataFrame that belongs to another node"
```

Comment:
We normally don't capitalize the message:
```suggestion
              "it is not possible to inspect a DataFrame that belongs to another node"
```

Reply:
There are two issues here:

1. We probably should not raise, otherwise it makes impossible to see the value in logs, IEx, etc

2. `df.data.resource` is accessing an implementation detail of the backend. Ideally we want to move this to `Explorer.DF.PolarsBackend` or similar. And instead of raising, maybe we just print something like:

```
#Explorer.DataFrame<        
  Polars[node: other@foobar]
>
```

WDYT?

Reply:
Yes, it makes sense

Reply:
Just to add a little bit: we have the column names and types. We could also display them if we want (without the values). WDYT?

Reply:
Ah, that's good. We can show them as:

```
Polars[node: ...]
foo s64 ???
bar string ???
```

Reply:
https://github.com/elixir-explorer/explorer/pull/784/commits/100de461e41385990d75c171851f7ae0a81f2ab7